app/Http/Controllers/Admin/AdminMemberController.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Exceptions\ApiException;
+use App\Http\Controllers\ApiController;
+use App\Http\Permissions\Permission;
+use App\Http\Permissions\PermissionDescribe;
+use App\Models\Member;
+use App\Services\Member\CreateMemberService;
+use App\Services\Member\UpdateMemberService;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Validation\Rule;
+use OpenApi\Attributes as OA;
+use Throwable;
+
+class AdminMemberController extends ApiController
+{
+    protected function initPermissions(): void
+    {
+        $this->permissionOneOf(Permission::globalAdmin);
+    }
+
+    #[OA\Post(
+        path: '/api/v1/admin/member',
+        description: new PermissionDescribe(Permission::globalAdmin),
+        summary: 'Create member',
+        requestBody: new OA\RequestBody(
+            content: new OA\JsonContent(
+                required: ['userId', 'facilityId' , 'hasFacilityAdmin'],
+                properties: [
+                    new OA\Property(property: 'userId', type: 'string', example: 'UUID'),
+                    new OA\Property(property: 'facilityId', type: 'string', example: 'UUID'),
+                    new OA\Property(property: 'hasFacilityAdmin', type: 'bool', example: true),
+                ]
+            )
+        ),
+        tags: ['Admin'],
+        responses: [
+            new OA\Response(response: 201, description: 'Created'),
+            new OA\Response(response: 400, description: 'Bad Request'),
+            new OA\Response(response: 401, description: 'Unauthorised'),
+        ]
+    )] /** @throws Throwable|ApiException */
+    public function post(Request $request, CreateMemberService $service): JsonResponse
+    {
+        $data = $request->validate([
+            'user_id' => [
```

Comment:
tutaj z jakiegoś powodu się nie sprawdził ten mechanizm trzymania walidatorów w modelu?

Reply:
tu chyba chodziło o kwestię przekazywania kilku pól naraz przy funkcji anonimowej, metoda `getInsertValidator` bierze tylko jedno pole naraz więc trzeba by to pewnie jakoś rozwinąć, nie zadziała to np. przy

```
                Rule::unique('members')->where(function ($query) use ($request) {
                    return $query->where('user_id', $request['user_id'])
                        ->where('facility_id', $request['facility_id']);
                }),
```